sonic_y_cable/y_cable_base.py

```diff
@@ -72,19 +70,33 @@ class YCableBase(Logger):
     PRBS_DIRECTION_GENERATOR = 1
     PRBS_DIRECTION_CHECKER = 2
 
-    def __init__(self, port, log_identifier):
+    def __init__(self, port, main_logger):
         """
         Args:
             port:
                  an Integer, the actual physical port connected to a Y cable
+            main_logger:
+                 a logging instance object, the main logging instance which can be used to log messages into syslog
+                 which can be called from derived classes using functions defined as log_info, log_warning etc.
+
         """
         self.port = port
+        self.main_logger = main_logger
```

Comment:
Again, suggest simplifying to `self.logger`. Can also name it `self._logger` to imply that it is private, so that no one tries to call it directly and only calls the `log_warning()`, etc. methods.

Reply:
fixed